src/Migration.php

```diff
@@ -0,0 +1,594 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Db\Migration;
+
+use Exception;
+use Throwable;
+use Yiisoft\Db\Connection\Connection;
+use Yiisoft\Db\Exception\InvalidConfigException;
+use Yiisoft\Db\Exception\NotSupportedException;
+use Yiisoft\Db\Query\Query;
+use Yiisoft\Db\Schema\ColumnSchemaBuilder;
+use Yiisoft\Db\Schema\SchemaBuilderTrait;
+use Yiisoft\Strings\StringHelper;
+
+use function implode;
+
+/**
+ * Migration is the base class for representing a database migration.
+ *
+ * Migration is designed to be used together with the "vendor/bin/yii migrate" command.
+ *
+ * Each child class of Migration represents an individual database migration which is identified by the child class
+ * name.
+ *
+ * Within each migration, the {@see up() method should be overridden to contain the logic for "upgrading" the database;
+ * while the {@see down()} method for the "downgrading" logic. The "yii migrate" command manages all available
+ * migrations in an application.
+ *
+ * If the database supports transactions, you may also override {@see safeUp()} and {@see safeDown()} so that if
+ * anything wrong happens during the upgrading or downgrading, the whole migration can be reverted in a whole.
+ *
+ * Note that some DB queries in some DBMS cannot be put into a transaction. For some examples, please refer to
+ * [implicit commit](http://dev.mysql.com/doc/refman/5.7/en/implicit-commit.html). If this is the case, you should still
+ * implement `safeUp()` and `safeDown()`, instead.
+ *
+ * Migration provides a set of convenient methods for manipulating database data and schema.
+ *
+ * For example, the {@see insert()} method can be used to easily insert a row of data into a database table; the
+ * {@see createTable()} method can be used to create a database table.
+ *
+ * Compared with the same methods in {@see Command}, these methods will display extra information showing the method
+ * parameters and execution time, which may be useful when applying migrations.
+ *
+ * For more details and usage information on Migration, see the [guide article on Migration](guide:db-migrations).
+ */
+class Migration implements MigrationInterface
+{
+    use SchemaBuilderTrait;
+
+    protected Connection $db;
+
+    /**
+     * @var int max number of characters of the SQL outputted. Useful for reduction of long statements and making
+     * console output more compact.
+     */
+    public int $maxSqlOutputLength = 0;
+
+    /**
+     * @var bool indicates whether the console output should be compacted.
+     *
+     * If this is set to true, the individual commands ran within the migration will not be output to the console.
+     * Default is false, in other words the output is fully verbose by default.
+     */
+    public bool $compact = false;
```

Comment:
Could it be done not in the form of public property?

Reply:
done.